### ISSUE TITLE
`isNumeric` used to determine whether value can be joined to an integer

### ISSUE THREAD

**Issue by schuemie** [OHDSI/WebAPI#2030]
### Expected behavior

No error on RedShfit.

### Actual behavior

Error on RedShift as described [here](https://github.com/OHDSI/SqlRender/issues/276)

In [this query](https://github.com/OHDSI/WebAPI/blob/master/src/main/resources/resources/cdmresults/sql/report/condition/drilldown/ageAtFirstOccurrence.sql#L14-L17) the `isNumeric` function is used to determine if `stratum_id_1` and `stratum_id_2` can be joined to `concept_id`, which I think is an integer. However, just because these numbers are numeric doesn't mean they're integers. This seems to throw an error under certain circumstances on RedShift.

**Comment by chrisknoll** [comment]
I think I'd like to change the approach on how we deal with the uncertainty of what the type is in `strata_` columns.

This is an example of our current work:
```
SELECT
  'Length of observation'                                  AS seriesName,
  xLengthOfObservation                                     AS xLengthOfObservation,
  round(1.0 * sum(ar2.count_value) / denom.count_value, 5) AS yPercentPersons
FROM (SELECT ar.*,
        cast(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) * 30 AS xLengthOfObservation
      FROM @results_database_schema.achilles_results ar WHERE analysis_id = 108) ar1
  INNER JOIN
  (
    SELECT *
    FROM @results_database_schema.achilles_results WHERE analysis_id = 108
  ) ar2 ON ar1.analysis_id = ar2.analysis_id AND cast(CASE WHEN isNumeric(ar1.stratum_1) = 1 THEN 
                                            ar1.stratum_1 ELSE null END AS INT) <= cast(CASE WHEN isNumeric(ar2.stratum_1) = 1 THEN ar2.stratum_1 ELSE null END AS INT)
  ,
  (
    SELECT count_value
    FROM @results_database_schema.achilles_results WHERE analysis_id = 1
  ) denom
GROUP BY xLengthOfObservation, denom.count_value
ORDER BY xLengthOfObservation ASC
```

The reason for the need of the CASE statements is that the WHERE clause is sometimes evaluated after all the rows are selected, and during selection we may do an improper cast of a string value that's not an integer.  The isNumeric() intends to short-circuit that but I'm thinking that we should look at the analysis_id column in the row to determine how the `strata_` column should be handled, and not blindly try to cast the nuermic value into an it.  The reasoning is that the analysis_id will be the authority on how the `strata_` column should be interpreted, and not just based on the format of the string.   Using the analysis_id to make this determination was the origional intent of using the WHERE clause, but we now know about the out-of-order application of the expressions used in SELECT and those used in WHERE.

There are 331 instances of the `isNumeric()` CASE statement in the code, so I do not want to make 331 changes if there is an objection to using the analysis_id to determine if the column should be returned as an INT vs. depending on isNumeric() (which will actually be changed into a `convert(int,convert(numeric,strata_1))` expression).

**Comment by schuemie** [comment]
@chrisknoll , just so I understand what you're saying, in [this query](https://github.com/OHDSI/WebAPI/blob/master/src/main/resources/resources/cdmresults/sql/report/condition/drilldown/ageAtFirstOccurrence.sql#L14-L17), instead of the join defined as 
```sql
ON CAST(CASE WHEN isNumeric(ard1.stratum_1) = 1 THEN ard1.stratum_1 ELSE null END AS INT) = c1.concept_id
```
it should be defined as 
```sql
ON CAST(CASE WHEN analysis_id = 406 THEN ard1.stratum_1 ELSE null END AS INT) = c1.concept_id
```
right?

**Comment by chrisknoll** [comment]
yes, I believe that will work.

**Comment by chrisknoll** [comment]
This has been implemented in WebAPI v2.13
